src/figuresClasses.ts

```diff
@@ -1,19 +1,96 @@
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
+}
+
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue'
+}
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape.Triangle = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('invalid side length');
+    }
+
+    const triangleSides = [a, b, c]
+      .sort((sideA: number, sideB: number) => sideB - sideA);
+    const isTriangleInvalid = triangleSides[0] >= (
+      triangleSides[1] + triangleSides[2]
+    );
 
+    if (isTriangleInvalid) {
+      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
+    }
```

Comment:
Minor, but you can skip creating extra variable and leave out the braces

```suggestion
    if (triangleSides[0] >= triangleSides[1] + triangleSides[2]) {
      throw new Error(`sides ${a}, ${b} and ${c} can't form a triangle`);
    }
```